### PR TITLE
refactor: move utils

### PR DESCRIPTION
**Pull request by platocrat**
Moved utils from `data-transport-layer/src/utils/common.ts` to `core-utils/src/common/common.ts`.

### CODE REVIEW

**Comment by platocrat** [comment]
final addition of utils from this recent merged PR: https://github.com/ethereum-optimism/data-transport-layer/pull/40

File: src/common/common.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  try {
+    if (condition() === false) {
+      throw new Error(`Assertion failed: ${reason}`)
+    }
+  } catch (err) {
+    throw new Error(`Assertion failed: ${reason}\n${err}`)
+  }
+}
+
+export const toRpcHexString = (n: number): string => {
```
Comment (smartcontracts) [inline src/common/common.ts:13]: @platocrat we could reopen this issue and get it merged if we only include the contents of this file.

Comment (smartcontracts) [reply src/common/common.ts:13]: Actually, even better would be to move `toRpcHexString` and `padHexString` to `common/hex-strings.ts`.

Comment (platocrat) [reply src/common/common.ts:13]: I'll reopen it rn

**Comment by platocrat** [comment]
Sorry, it took me so long!

I messed a _ton_ of stuff up and had to figure out how to resolve it. 
Ended up git hard resetting back to my most recent commit (which was somehow painful to find 😅) in this PR

**Comment by platocrat** [comment]
What I was trying to proactively fix was the merge conflict when the merge is attempted. 

I failed by trying to clone a recent copy of `ethereum-optimism/core-utils`, change the remote URL to my fork, checkout of a new version of the `refactor/move-utils` branch, and copy over all my changes from this PR and push that. 

Didn't work out well for various reasons...

**Comment by platocrat** [comment]
That merge fixes everything. Sorry again for taking so long 😕

**Review by smartcontracts — COMMENTED** [review]
LGTM. Will ask for review from @tynes before we merge.

**Review by tynes — APPROVED** [review]

**Pull request merged**